### ISSUE TITLE
replace variable in a greta array

### ISSUE THREAD

**Issue by goldingn** [greta-dev/greta#196]
This errors:
```r
x <- normal(0, 1, dim = 3)
x[1] <- 0
```
> ```
>  Error: cannot replace values in a variable greta array 
> ```

But it needn't, we're just creating a new greta array `x` which is an operation.

I'm not sure whether allowing this would make for a clearer or a less clear interface. The first random variable will still be floating around, and its density will be included in the joint model (though as an unconnected independent variable).

This issue is just a reminder to think about this in the future. Any input very welcome!

**Comment by rdpeng** [comment]
So, I ran into this issue the other day because I was trying to use greta for a missing data problem. I wanted to create a greta variable array with a multivariate Normal distribution and then just specify that some of the values are observed (and therefore do not need to be imputed). That said, I don't know if this is the proper way to specify a missing data problem with greta, but at first it seemed to me like it would be.

**Comment by goldingn** [comment]
Yeah, that's one case where there's not an obvious workaround that can be achieved by combining greta arrays. I think that would be possible with the planned extension in #117, which will hopefully end up in an extension package before too long. Though I'd really like someone else to take that on (long to-do list), so do let me know if you have any students keen for a stats/software project! 😃
